mica-core/src/main/java/org/obiba/mica/config/AsyncConfiguration.java

```diff
@@ -52,15 +52,25 @@ public AsyncUncaughtExceptionHandler getAsyncUncaughtExceptionHandler() {
     return new SimpleAsyncUncaughtExceptionHandler();
   }
 
-  @Bean
   @Override
+  @Bean
   public Executor getAsyncExecutor() {
-    log.debug("Creating Async Task Executor");
     ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
     executor.setCorePoolSize(propertyResolver.getProperty("corePoolSize", Integer.class, DEFAULT_POOL_SIZE));
     executor.setMaxPoolSize(propertyResolver.getProperty("maxPoolSize", Integer.class, DEFAULT_MAX_POOL_SIZE));
     executor.setQueueCapacity(propertyResolver.getProperty("queueCapacity", Integer.class, DEFAULT_QUEUE_CAPACITY));
     executor.setThreadNamePrefix("mica-executor-");
     return new ExceptionHandlingAsyncTaskExecutor(executor);
   }
+
+  @Bean(name="opalExecutor")
+  public Executor getOpalAsyncExecutor() {
+    log.debug("Creating Async Task Executor");
+    ThreadPoolTaskExecutor executor = new ThreadPoolTaskExecutor();
+    executor.setCorePoolSize(propertyResolver.getProperty("opal.corePoolSize", Integer.class, DEFAULT_POOL_SIZE));
+    executor.setMaxPoolSize(propertyResolver.getProperty("opal.maxPoolSize", Integer.class, DEFAULT_MAX_POOL_SIZE));
+    executor.setQueueCapacity(propertyResolver.getProperty("queueCapacity", Integer.class, DEFAULT_QUEUE_CAPACITY));
+    executor.setThreadNamePrefix("mica-opal-executor-");
+    return new ExceptionHandlingAsyncTaskExecutor(executor);
```

Comment:
accepting this PR until a better executor strategy is tested. ideally should have bounded queue size and increase threads before the queue is full.